### ISSUE TITLE
Transformation Descriptor/Ambient context, and Skew

### ISSUE THREAD

**Issue by flew2bits** [QuestPDF/QuestPDF#514]
Issue #502 is what led me to this request. 

In Illustrator, and likely other drawing programs, you can specify one of 9 cardinal points about which to apply transformations, effectively moving the origin of the container. They are top-left, top-center, top-right, left-center, center, right-center, bottom-right, bottom-center, and bottom-left. 

I propose adding a transformation element to containers that can take a transformation descriptor. It would work similarly:

```c#
    container.Transform(transform =>
        transform.About(TransformAbout.Center);
        transform.SkewX(10);
        transform.Rotate(45);
        transform.Scale(1.2f);
    );
```

Since the renderer has the container size, the `About` descriptor would tell the transformer to pre- and post-translate the container to the 0, 50, or 100% height and width points, effectively changing the transformation origin, and then applying the other transformations in order.

This would also be another possible place to add an ambient context ala DefaultTextStyle and DefaultParagraphStyle (see #513). I think it would only make sense to have the `About` option be in the ambient context, but this isn't really a necessity.

From the example, I'd also like to see Skew added as a transformation.

It's possible to apply these types of transformations currently, but you have to apply them to a known, fixed-sized container, mainly the page. If you want to transform something several containers deep around any point other than the top-left, you have to perform multiple transformations to make it happen because you have to "unapply" transformation up the hierarchy.

I don't believe this would be too difficult to implement. It would require a new Descriptor type, but transforming around a different center simply requires two additional translate operations.

**Comment by MarcinZiabek** [comment]
Do you think that this PR could help? https://github.com/QuestPDF/QuestPDF/pull/357 This is my old idea about giving more freedom when performing transformations, especially when using containers with unknown sizes.

**Comment by flew2bits** [comment]
I don't think so, entirely. I'd have to download the PR and try it out, but based on my quick perusal it addresses some of the issues, it's not general purpose enough to handle some of the cases I addressed. That said, I think both features have utility. I'm not immediately sure that my proposal would be able to implement all of the functionality that the PR addresses.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

QuestPDF currently provides individual transformation elements — `Rotate`, `Scale`, `Offset`, and `Flip` — but a unified transform descriptor with a configurable origin (the cardinal-point `About` option) and a skew transformation are not yet available. Improved transformation functionality covering these scenarios is on our backlog, and we plan to expand it in the future.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
